pkg/kubelet/lifecycle/handlers.go

```diff
@@ -132,6 +133,17 @@ func (hr *handlerRunner) runHTTPHandler(pod *v1.Pod, container *v1.Container, ha
 		}
 		resp, err := hr.httpDoer.Do(req)
 		discardHTTPRespBody(resp)
+
+		if err != nil {
+			if urlerr, ok := err.(*url.Error); ok && urlerr.Err.Error() == "http: server gave HTTP response to HTTPS client" {
+				// TODO: emit an event about the fallback
+				// TODO: increment a metric about the fallback
+				klog.V(1).ErrorS(err, "Falling back to HTTP.", "pod", klog.KObj(pod))
+				req.URL.Scheme = "http"
+				resp, err = hr.httpDoer.Do(req)
```

Comment:
I'd suggest something like this:

```suggestion
				httpResp, httpErr := hr.httpDoer.Do(httpReq)
				discardHTTPRespBody(httpResp)
				if httpErr == nil {
				  // The fallback succeeded, don't return an error
				  err = nil
				}
```

this way, if _both_ the https and http request get errors, we return the initial error from talking to the https endpoint as they specified